src/components/gauge/gauge.tsx

```diff
@@ -0,0 +1,445 @@
+                      this.calcThickness() / 2 * (this.minorSize / 100)) /
+                      8 +
+                    "px",
+                  "mix-blend-mode": "difference"
+                }}
+              >
+                {(calcPercentage(this) > 100
+                  ? 100
+                  : calcPercentage(this) < 0
+                    ? 0
+                    : Math.round(calcPercentage(this))) + "%"}
+              </div>
+            ) : (
+              ""
+            )}
+          </div>
+        </div>
+      );
+    } else if (this.gaugeType === "Line") {
+      const divRanges = [];
+      const divRangesName = [];
+      let currentMargin = 0;
+      let currentMargin2 = 0;
+      //////////////////////////////////////////
+      function calcPositionRange(preValue) {
+        currentMargin += preValue;
+        return currentMargin;
+      }
+      function calcPositionName(preValue) {
+        currentMargin2 += preValue;
+        return currentMargin2;
+      }
+      //////////////////////////////////////////
+      for (let i = childRanges.length - 1; i >= 0; i--) {
+        divRanges.push(
+          <div
+            class="range"
+            style={{
+              "background-color": childRanges[i].color,
+              "box-shadow": !this.styleShadow ? "none" : "",
+              "margin-left":
+                calcPositionRange(
+                  !!childRanges[i + 1]
+                    ? parseInt(childRanges[i + 1].getAttribute("amount"), 10) *
+                      100 /
+                      this.totValues
+                    : 0
+                ) + "%",
+              width:
+                parseInt(childRanges[i].getAttribute("amount"), 10) *
+                  100 /
+                  this.totValues +
+                "%"
+            }}
+          />
+        );
+        divRangesName.push(
+          <span
+            class="rangeName"
+            style={{
+              "margin-left":
+                calcPositionName(
+                  !!childRanges[i + 1]
+                    ? parseInt(childRanges[i + 1].getAttribute("amount"), 10) *
+                      100 /
+                      this.totValues
+                    : 0
+                ) + "%",
+              width:
+                parseInt(childRanges[i].getAttribute("amount"), 10) *
+                  100 /
+                  this.totValues +
+                "%"
+            }}
+          >
+            {childRanges[i].name}
+          </span>
+        );
+      }
+      divRanges.reverse();
+      divRangesName.reverse();
+      return (
+        <div
+          class="gaugeContainerLine"
+          style={{
+            border: this.styleBorder
+              ? this.styleBorderWidth + "px solid " + this.styleBorderColor
+              : "",
+            height: 5 * this.calcThickness() + "px"
+          }}
+        >
+          <div class="rangesContainer">
+            {divRanges.map(gaugeRange => gaugeRange)}
```

Comment:
This would be the same as `{divRanges}`, right?
There's no need for the `map` call.

```suggestion
            {divRanges}
```

Reply:
I did not know that! 😲